### PR TITLE
Support signing TypedData

### PR DESCRIPTION
**Pull request by cptartur**
## Describe your changes

This PR adds support for signing TypedData both in `Account` and `Signer` interfaces

## Linked issues

Closes

## Breaking changes

- [x] This issue contains breaking changes

* Account interface added `signTypedData` and `verifyTypedDataSignature`
* Signer interface added `signTypedData`

### CODE REVIEW

File: lib/src/main/kotlin/com/swmansion/starknet/signer/Signer.kt [hunk 1]
```diff
@@ -15,10 +16,16 @@ interface Signer {
...
-    // TODO: add more params
     fun signTransaction(transaction: Transaction): Signature
 
+    /**
+     * Sign TypedData
+     *
+     * @param typedData TypedData instance to sign
+     * @return a signature of provided typedData
+     */
+    fun signTypedData(typedData: TypedData, accountAddress: Felt): Signature
```
Comment (bartekryba) [inline lib/src/main/kotlin/com/swmansion/starknet/signer/Signer.kt:27]: Missing param comment for accountAddress

File: lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt [hunk 2]
```diff
@@ -76,6 +77,23 @@ interface Account {
...
+
+    /**
+     * Verify a signature of TypedData on StarkNet
+     *
+     * @param typedData a TypedData instance which signature will be verified
+     * @param signature a signature of typedData
+     * @return `true` if signature is valid, `false` otherwise
+     */
+    fun verifyTypedDataSignature(typedData: TypedData, signature: Signature): Request<Boolean>
+
```
Comment (THenry14) [inline lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt:96]: why not just `sign` and `verify`?

File: lib/src/main/kotlin/com/swmansion/starknet/account/StandardAccount.kt [hunk 3]
```diff
@@ -106,6 +109,45 @@ class StandardAccount(
...
+                        return@handle handleValidationError(exception)
+                    }
+                    return@handle result[0] > Felt.ZERO
+                }
+            }
+        }
+    }
+
+    private fun handleValidationError(e: RequestFailedException): Boolean {
+        val regex = """Signature\s.+,\sis\sinvalid""".toRegex()
```
Comment (THenry14) [inline lib/src/main/kotlin/com/swmansion/starknet/account/StandardAccount.kt:144]: Maybe add some comment about what this regex does?

File: lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt (hunk 2 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt:96]: I wanted to make `signTypedData` a distinct method, similarly to `signDeclare` etc. `sign` is already used for InvokeTransactions.

`verifyTypedDataSignature` followed that convention.

**Review by THenry14 — APPROVED** [review]

**Pull request merged**